formats/vector-to-label-plugin/src/polus/plugins/formats/vector_to_label/dynamics/vector_to_label.py

```diff
@@ -0,0 +1,290 @@
+    resolved = new_labels > -1
+
+    while numpy.any(new_labels != old_labels):
+        old_labels = new_labels
+        resolved = new_labels > -1
+        orig_ind = tuple(p[resolved] for p in original_index)
+        flow_ind = tuple(p[resolved] for p in flow_index)
+        labels[orig_ind] = labels[flow_ind]
+        new_labels = labels[flow_index]
+
+    return tuple(o[~resolved] for o in original_index), flow_points[:, ~resolved]
+
+
+def interpolate_flow(vectors: numpy.ndarray, points: numpy.ndarray) -> numpy.ndarray:
+    """Move a pixel to a new location based on the current vectors in the region.
+
+    This function takes a vector field and a set of points, and moves each point one
+    step along the vector field. Points exist in continuous space, and the value of the
+    vector field at a point is determined by interpolating local vector values at the
+    surrounding pixels.
+
+    This function works on n-dimensional tensor fields.
+
+    Args:
+        vectors (numpy.ndarray): n-dimensional tensor field
+        points (numpy.ndarray): An array of points
+
+    Returns:
+        numpy.ndarray: The points after following flows for one step
+    """
+    ndims = vectors.shape[0]
+
+    points_floor = points.astype(numpy.int32)
+    points_ceil = points_floor + 1
+    for d in range(ndims):
+        points_ceil[d, :] = numpy.clip(
+            points_ceil[d, :],
+            a_min=0,
+            a_max=vectors.shape[d + 1] - 1,
+        )
+
+    points_norm = points - points_floor
+    points_norm_inv = numpy.clip(
+        points_ceil - points_floor - points_norm,
+        a_min=0,
+        a_max=None,
+    )
+
+    # Shape of the local points should be:
+    # N x P x (2, ) * N
+    # where N is the number of dimensions and P is the number of points
+    shape = points.shape + (2,) * ndims
+    vector_field = numpy.zeros(shape)
+    for index in itertools.product(range(2), repeat=ndims):
+        vector_field[(slice(None), slice(None), *index)] = vectors[
+            (
+                slice(None),
+                *tuple(
+                    points_floor[d] if i == 0 else points_ceil[d]
+                    for d, i in enumerate(index)
+                ),
+            )
+        ]
+
+    # Run interpolation
+    for d in reversed(range(ndims)):
+        if vector_field.ndim == 3:  # noqa: PLR2004
+            p = numpy.zeros((*vector_field.shape, 1))
+            vector_field = numpy.expand_dims(vector_field, axis=2)
+        else:
+            p = numpy.zeros(vector_field.shape[:-1] + (1,))
+
+        shape = (1, points_norm.shape[1]) + (1,) * (vector_field.ndim - 4)
+        p[..., 0, 0] = points_norm_inv[d : d + 1].reshape(shape)
+        p[..., 1, 0] = points_norm[d : d + 1].reshape(shape)
+
+        # Stacked matrix multiply
+        vector_field = vector_field @ p
+
+        vector_field = vector_field.squeeze(axis=-1)
+
+    return vector_field[..., 0] + points
+
+
+def convert(
+    vectors: numpy.ndarray,
```

Comment:
are training data from cell pose also providing the binary masks? So the problem to be solved in only focused on determining boundaries between objects?